packages/front-end/src/app/components/EGLabView.vue

```diff
@@ -182,11 +184,25 @@
   ];
 
   function viewRunDetails(run: GenericRun) {
-    $router.push({ path: `/labs/${props.labId}/${run.type}-run/${run.id}`, query: { tab: 'Run Details' } });
+    const foundRun = getLabRunId(run.id);
+
+    if (foundRun) {
+      $router.push({
+        path: `/labs/${props.labId}/${run.type}-run/${run.id}`,
+        query: { tab: 'Run Details', runId: foundRun.RunId },
+      });
+    }
   }
 
   function viewRunResults(run: GenericRun) {
-    $router.push({ path: `/labs/${props.labId}/${run.type}-run/${run.id}`, query: { tab: 'Run Results' } });
+    const foundRun = getLabRunId(run.id);
+
+    if (foundRun) {
+      $router.push({
+        path: `/labs/${props.labId}/${run.type}-run/${run.id}`,
+        query: { tab: 'Run Results', runId: foundRun.RunId },
```

Comment:
requiring the (laboratory) `runId` to be passed as a query param will mean that the Run's page is coupled to this click event on the previous page. undoing this pattern has been the cause of a lot of our refactoring time.

the pattern I think we need to follow is to load Laboratory Runs into a store and look them up on the Run page, same as what we do for orgs, labs, pipelines/workflows, and runs.

@simondib I'm aware that it's about 4pm on your last day for the year - I'm happy to get this over the line tomorrow if that's good with you

Reply:
Hmm, is this a major issue? The page isn't navigable outside of the row link and it avoids having to deal with store state. 

(Having said that I might be overlooking the negative impact of this pattern.. my mental energy is getting very low at this point of the day/week/year 😅 So I'd be very grateful if you could take the refactor on).

Reply:
yep all good 👍 

more than a couple of my big refactor tasks have been a flow being built around one entry point and then needing reworking to enable another. my other PR that's open right now is one of these, where the setup for a run is done on click of the pipeline row - then the Run Another button reloads the page but that setup on row click wasn't there, so it had to be pulled into the run page. 

in the longer run these Laboratory Runs should replace the SeqeraRun / OmicsRuns that we already store cache for this reason, so it should end up with a store of its own anyway. might as well do it tomorrow :)

Reply:
Perfect, sounds like a worthwhile change. Thanks for the explanation and for taking it on 👍